extension/js/common/core/crypto/pgp/msg-util.ts

```diff
@@ -306,6 +306,33 @@ export class MsgUtil {
     return diagnosis;
   }
 
+  public static isPasswordMesageEnabled(subject: string, disallowTerms: string[]) {
+    if (!subject || !Array.isArray(disallowTerms) || disallowTerms.length === 0) {
+      return true; // Allow by default if no terms are specified or subject is empty
+    }
+
+    // Normalize subject for case-insensitive comparison
+    const lowerCaseSubject = subject.toLowerCase();
+
+    // Check if any disallow term exists as an exact match in the subject
+    for (const term of disallowTerms) {
+      const lowerCaseTerm = term.toLowerCase();
+
+      // Check for exact matches (full-term match within the subject)
+      if (
+        lowerCaseSubject.includes(lowerCaseTerm) &&
+        (lowerCaseSubject === lowerCaseTerm ||
+          lowerCaseSubject.startsWith(lowerCaseTerm + ' ') ||
+          lowerCaseSubject.endsWith(' ' + lowerCaseTerm) ||
+          lowerCaseSubject.includes(' ' + lowerCaseTerm + ' '))
+      ) {
+        return false;
+      }
```

Comment:
disallowed terms can be surrounded not only by spaces.
for example, term can be `internal data` and subject string `[internal data] year results` should trigger error message, as it contains full `internal data` string.
probably using regex should make it simpler to detect such cases

Reply:
Nice catch 👍 
Let me take a look